packages/runtime-dom/src/apiCustomElement.ts

```diff
@@ -520,7 +520,9 @@ export class VueElement
   }
 
   private _update() {
-    render(this._createVNode(), this._root)
+    const vnode = this._createVNode()
+    if (this._app) vnode.appContext = this._app._context
```

Comment:
The reason for checking `this._app` here is that the test case will report an error. https://github.com/vuejs/core/blob/b7b093203775d292ede1ec5841452af9e2a7f3cc/packages/runtime-dom/__tests__/customElement.spec.ts#L137-L147
When re-inserting, `this._app` is `null` because we cleared it in `disconnectedCallback`.
For custom elements that remove and then re-insert (`this._resolved = true`), just calling `_update` is not enough. https://github.com/vuejs/core/blob/2235643a4e1c559fea25ae1cfb9f6058ac8becfc/packages/runtime-dom/src/apiCustomElement.ts#L300-L303
We must also reinitialize `this._ob` because `disconnectedCallback` clears `this._ob`. 
For more edge cases refer to #12412. Maybe once #12413 is merged, the check for `this._app` can be removed.